src/compiler/types.ts

```diff
@@ -1780,12 +1863,30 @@ namespace ts {
         kind: SyntaxKind.ClassExpression;
     }
 
-    export interface ClassElement extends RealDeclaration {
+    export interface ClassElement extends DeclarationBase {

```

Comment:
I think we should steer clear from large discriminated unions, including this one for `Declaration`. Is there another way we can structure it? Something like:

```ts
interface Declaration extends Node {
  _declarationBrand: any;  
}
interface DeclarationWithName extends Declaration {
  name?: DeclarationName;
}
```

Have the expressions inherit from `Declaration` and have other named declarations inherit from `DeclarationWithName`?